app/src/androidTest/java/com/example/triptracker/e2eTest.kt

```diff
@@ -0,0 +1,147 @@
+          }
+        }
+      }
+    }
+
+    composeTestRule.onNodeWithTag("ProfileOverview").assertIsDisplayed()
+
+    // travel to the friends screen
+    composeTestRule.onNodeWithTag("FriendsButton").assertHasClickAction()
+    composeTestRule.onNodeWithTag("FriendsButton").performClick()
+
+    // verify that the favorites screen is open
+    composeTestRule.onNodeWithTag("FriendsFinderScreen").assertIsDisplayed()
+
+    // Go back and assert we are in the profile screen
+    composeTestRule.onNodeWithTag("GoBackButton").performClick()
+    composeTestRule.onNodeWithTag("ProfileOverview").assertIsDisplayed()
+
+    // travel to the favorites
+    composeTestRule.onNodeWithTag("FavoritesButton").assertHasClickAction()
+    composeTestRule.onNodeWithTag("FavoritesButton").performClick()
+
+    // verify that the favorites screen is open
+    composeTestRule.onNodeWithTag("UserProfileFavouriteScreen").assertIsDisplayed()
+
+    // Go back and assert we are in the profile screen
+    composeTestRule.onNodeWithTag("GoBackButton").performClick()
+    composeTestRule.onNodeWithTag("ProfileOverview").assertIsDisplayed()
+
+    // travel to the settings
+    composeTestRule.onNodeWithTag("SettingsButton").assertHasClickAction()
+    composeTestRule.onNodeWithTag("SettingsButton").performClick()
+
+    // verify that the followers screen is open
+    composeTestRule.onNodeWithTag("UserProfileSettings").assertIsDisplayed()
+
+    // Change privacy to private
+    composeTestRule.onNodeWithTag("ProfilePrivacyButton").performClick()
+
+    // Go back and assert we are in the profile screen
+    composeTestRule.onNodeWithTag("GoBackButton").performClick()
+    composeTestRule.onNodeWithTag("ProfileOverview").assertIsDisplayed()
+
+    // travel to the my trips
+    composeTestRule.onNodeWithTag("MyTripsButton").assertHasClickAction()
+    composeTestRule.onNodeWithTag("MyTripsButton").performClick()
+
+    // verify that the my trips screen is open
+    composeTestRule.onNodeWithTag("UserProfileMyTripsScreen").assertIsDisplayed()
+
+    // Go back and assert we are in the profile screen
+    composeTestRule.onNodeWithTag("GoBackButton").performClick()
+    composeTestRule.onNodeWithTag("ProfileOverview").assertIsDisplayed()
+
+    // Edit profile
+    composeTestRule.onNodeWithContentDescription("Edit").assertHasClickAction()
+    composeTestRule.onNodeWithContentDescription("Edit").performClick()
+
+    // verify that the edit screen is open
+    composeTestRule.onNodeWithTag("UserProfileEditScreen").assertIsDisplayed()
+
+    // input into the field
+    composeTestRule.onNodeWithText("Save").isDisplayed()
+
+    // Go back and assert we are in the profile screen
+    composeTestRule.onNodeWithTag("GoBackButton").performClick()
+    composeTestRule.onNodeWithTag("ProfileOverview").assertIsDisplayed()
+  }
+
+  /*
+   Test to check the flow of the app navigating through different places, to be implemented
+  */
+  @Test fun appFlow() {}
```

Comment:
What is this function for, i dont really understand, what is that doing here

Reply:
I just planned on implementing that later, it's supposed to be navigating test through the screens